runtime-transaction/src/runtime_transaction.rs

```diff
@@ -38,7 +38,9 @@ impl StaticMeta for RuntimeTransactionStatic {
     }
 }
 
-impl RuntimeTransactionStatic {
+impl<M> DynamicMeta for RuntimeTransaction<M> {}
```

Comment:
This seems to defeat some of our previously introduced type-safety. Now DynamicMeta is **always** available regardless of the message type. The dynamic meta might not have been loaded yet, if `M` is a `SanitizedVersionedMessage` for example.

Could add a trait bound on M for this impl:
```
trait DynamicSupportingMessage {}

impl<M: DynamicSupportingMessage> for RuntimeTransaction<M> {}
impl DynamicSupportingMessage for SanitizedMessage {}
```

Reply:
3344e0c implements static/dynamic meta traits for specialized transaction types. It's bit verbose, but feel easier to manage at this stage of dev. wdyt?

Reply:
I'm just not a huge fan of duplicating the access code, even if it is really simple. As we add more metadata, or support different message types, we have to keep copy-pasting which will leave more chance for error imo.

wdyt of the following (tested and will compile as drop in replacement for your impls):
```rust
// These traits gate access to static and dynamic metadata
// so that only transactions with supporting message types
// can access them.
trait StaticMetaAvailable {}
trait DynamicMetaAvailable: StaticMetaAvailable {}

// Implement the gate traits for the message types that should
// have access to the static and dynamic metadata.
impl StaticMetaAvailable for SanitizedVersionedMessage {}
impl StaticMetaAvailable for SanitizedMessage {}
impl DynamicMetaAvailable for SanitizedMessage {}

impl<M: StaticMetaAvailable> StaticMeta for RuntimeTransaction<M> {
    fn message_hash(&self) -> &Hash {
        &self.meta.message_hash
    }
    fn is_simple_vote_tx(&self) -> bool {
        self.meta.is_simple_vote_tx
    }
}

impl<M: DynamicMetaAvailable> DynamicMeta for RuntimeTransaction<M> {
    fn i_am_dynamic(&self) -> bool {
        true
    }
}
```

Reply:
As I submitted comment, I think the traits should probably be:
```rust
StaticMetadataAccess
DynamicMetadataAccess
```
instead of `Available`

Reply:
dropped your suggestion at df369e9. Tried to avoid additional `*Access` traits by `struct RuntimeTransaction<M, MetaType>`, It isn't as elegant as your solution. Thank you!